src/app/clusters/mode-select-server/mode-select-server.cpp

```diff
@@ -15,118 +16,198 @@
  *    limitations under the License.
  */
 
-#include <app-common/zap-generated/att-storage.h>
 #include <app-common/zap-generated/attributes/Accessors.h>
-#include <app-common/zap-generated/cluster-objects.h>
-#include <app-common/zap-generated/ids/Attributes.h>
-#include <app-common/zap-generated/ids/Clusters.h>
-#include <app/AttributeAccessInterface.h>
-#include <app/CommandHandler.h>
-#include <app/ConcreteCommandPath.h>
-#include <app/clusters/mode-select-server/supported-modes-manager.h>
-#include <app/util/af.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/mode-select-server/mode-select-server.h>
+#include <app/clusters/on-off-server/on-off-server.h>
 #include <app/util/attribute-storage.h>
-#include <app/util/config.h>
-#include <app/util/error-mapping.h>
-#include <app/util/odd-sized-integers.h>
 #include <app/util/util.h>
-#include <lib/support/CodeUtils.h>
 #include <platform/DiagnosticDataProvider.h>
 
-#ifdef EMBER_AF_PLUGIN_ON_OFF
-#include <app/clusters/on-off-server/on-off-server.h>
-#endif // EMBER_AF_PLUGIN_ON_OFF
-
-using namespace std;
+// using namespace std;
 using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::ModeSelect;
-using namespace chip::Protocols;
 using chip::Protocols::InteractionModel::Status;
 
 using BootReasonType = GeneralDiagnostics::BootReasonEnum;
 
-static InteractionModel::Status verifyModeValue(const EndpointId endpointId, const uint8_t newMode);
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ModeSelect {
 
-namespace {
+// todo find a cleaner solution by modifying the zap generated code.
+EmberAfStatus Instance::GetFeature(uint32_t * value) const
```

Comment:
This code in written similar to the zap generated accessor functions because it is hoped that we will eventually have this kind of generic accessor function generated by zap for parent/primary clusters.

Reply:
yes, but the ZAP generated code is basically the only place that uses this and it's legacy Ember usage that has is adaptered elsewhere.

The actual idiomatic way in cluster services is `chip::Protocols::InteractionModel::Status`.

Examples in SDK:
![image](https://github.com/project-chip/connectedhomeip/assets/1313847/10bb7650-bc24-4aa9-94e6-06b652c2e52d)